x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/math/BinaryOptionalNumericFunction.java

```diff
@@ -88,7 +89,7 @@ public Expression replaceChildren(List<Expression> newChildren) {
     @Override
     public ScriptTemplate asScript() {
         ScriptTemplate leftScript = asScript(left);
-        ScriptTemplate rightScript = asOptionalScript(right);
+        ScriptTemplate rightScript = asScript(right == null ? Literal.NULL : right);
```

Comment:
Personal preference: I kinda liked more the method `asOptionaScript`

Reply:
I removed it since it was not used anywhere outside the base class.
The script generation needs improvement anyway as it has evolved beyond the initial requirements.